service/api.go

```diff
@@ -676,6 +756,8 @@ func (as *APIServer) GetServiceApp() *gimlet.APIApp {
 	app.Route().Version(2).Route("/task/{taskId}/parser_project").Wrap(requireTaskSecret).Handler(as.GetParserProject).Get()
 	app.Route().Version(2).Route("/task/{taskId}/project_ref").Wrap(requireTaskSecret).Handler(as.GetProjectRef).Get()
 	app.Route().Version(2).Route("/task/{taskId}/expansions").Wrap(requireTask, requireHost).Handler(as.GetExpansions).Get()
+	app.Route().Version(2).Prefix("/task/{taskId}").Route("/new_push").Wrap(requireTask).Handler(as.NewPush).Post()
```

Comment:
Last thing: I think both of these should use requireTaskSecret rather than requireTask, since it's a stricter check and we are passing the secret in.

Reply:
Also can this just use .Route instead of .Prefix, just for consistency with the other agent routes?